src/module.c

```diff
@@ -1702,8 +1689,10 @@ int RM_ReplyWithNull(RedisModuleCtx *ctx) {
 int RM_ReplyWithCallReply(RedisModuleCtx *ctx, RedisModuleCallReply *reply) {
     client *c = moduleGetReplyClient(ctx);
     if (c == NULL) return REDISMODULE_OK;
-    sds proto = sdsnewlen(reply->proto, reply->protolen);
-    addReplySds(c,proto);
+    size_t proto_len;
+    const char* proto = callReplyGetProto(reply, &proto_len);
+    sds proto_dup = sdsnewlen(proto, proto_len);
+    addReplySds(c,proto_dup);
```

Comment:
we can call `addReplyProto` and save the extra mallc,free,memcpy
```suggestion
    addReplyProto(c,proto, proto_len);
```